dask/array/core.py

```diff
@@ -2649,15 +2682,48 @@ def elemwise(op, *args, **kwargs):
                                                   tokenize(op, dt, *args))
 
     if other:
-        return atop(partial_by_order, expr_inds,
+        result = atop(partial_by_order, expr_inds,
                     *concat((a, tuple(range(a.ndim)[::-1])) for a in arrays),
                     dtype=dt, name=name, function=op, other=other,
                     token=funcname(op))
     else:
-        return atop(op, expr_inds,
+        result = atop(op, expr_inds,
                     *concat((a, tuple(range(a.ndim)[::-1])) for a in arrays),
                     dtype=dt, name=name)
 
+    return handle_out(out, result)
+
+
+def handle_out(out, result):
```

Comment:
I'm somewhat cautious about using decorators for internal things like this.  I think that we're capable of tracking down the few corner cases here and I would be concerned about reducing the explicit-ness of the code.  This is pretty subjective though and just a personal preference.